### PR TITLE
(flw 2471) keyboard adoptable

### PR DESCRIPTION
**Pull request by Viscount88**

### CODE REVIEW

File: fearless/Common/Protocols/KeyboardAdoptable.swift [hunk 1]
```diff
@@ -14,21 +14,16 @@ extension KeyboardAdoptable {
...
         keyboardHandler = nil
     }
 }
 
-protocol KeyboardViewAdoptable: KeyboardAdoptable {
-    var target: UIView? { get }
-    var currentKeyboardFrame: CGRect? { get set }
-    var shouldApplyKeyboardFrame: Bool { get }
+protocol KeyboardViewAdoptable: KeyboardAdoptable, KeyboardHandlerDelegate {
+    var target: Constraint? { get }
```
Comment (DRadmir) [inline fearless/Common/Protocols/KeyboardAdoptable.swift:26]: what if I use frame for view? I can use offsetFromKeyboardWithInset?

Comment (DRadmir) [reply fearless/Common/Protocols/KeyboardAdoptable.swift:26]: And we can't open pull request in SoraFoundation, because independent from SnapKit

**Review by DRadmir — CHANGES_REQUESTED** [review]

File: fearless/Common/Helpers/FearlessKeyboardHandler.swift [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+    var handleWillShow: KeyboardFrameChangeHandleBlock?
+    var handleWillHide: KeyboardFrameChangeHandleBlock?
+
+    // MARK: Initialization
+
+    deinit {
+        NotificationCenter.default.removeObserver(self)
+    }
+
+    public convenience init() { self.init(with: nil) }
```
[the remark this answers is not in the document]
Comment (Viscount88) [reply fearless/Common/Helpers/FearlessKeyboardHandler.swift:18]: we have no delegate, but handlers for keyboard still working

File: fearless/Common/Protocols/KeyboardAdoptable.swift (hunk 1 above)
Comment (Viscount88) [reply fearless/Common/Protocols/KeyboardAdoptable.swift:26]: KeyboardAdoptable is not part of SoraFoundation, only KeyboardHandler. Use constraint is more safety and obvious than uiview

**Pull request merged**